test-case/volume-basic-test.sh

```diff
@@ -39,41 +39,63 @@ maxloop=${OPT_VAL['l']}
 
 start_test
 
-func_error_exit()
+check_alsa_tool_process()
 {
-    dloge "$*"
-    pkill -9 aplay
-    exit 1
+    sleep 1
+    if [[ "$SOF_ALSA_TOOL" = "alsa" ]]; then
+        # Check if the aplay process is running
+        if [[ ! $(pidof aplay) ]]; then
+            die "aplay process is terminated too early"
+        fi
+    elif [[ "$SOF_ALSA_TOOL" = "tinyalsa" ]]; then
+        # Check if the tinyplay process is running
+        if [[ ! $(pidof tinyplay) ]]; then
+            die "tinyplay process is terminated too early"
+        fi
+    else
+        echo "Unknown alsa tool $SOF_ALSA_TOOL"
+    fi
+}
+
+func_exit()
+{
+
+    if [[ "$SOF_ALSA_TOOL" = "alsa" ]]; then
+        pkill -9 aplay
+    elif [[ "$SOF_ALSA_TOOL" = "tinyalsa" ]]; then
+        pkill -9 tinyplay
+    else
+        echo "Unknown alsa tool $SOF_ALSA_TOOL"
+    fi
+
+    if [[ "$1" -eq 1 ]]; then
```

Comment:
All SOF_ALSA_TOOL logic should be in the same central place: `lib.sh`. I suspect it should also search for `arecord`

`func_exit 1` is cryptic and error-prone. It's much simpler and clearer to have two distinct functions. Then every test can do what it wants and it's always clear what it does.

```shell
kill_play_record()
{
  if "$SOF_ALSA_TOOL" 
     kill ...
}

cleanup_and_fail() # better name if you can think of one
{
   kill_play_record
   die "$1"
}

```